pyscf_ipu/electron_repulsion/direct.py

```diff
@@ -39,8 +39,10 @@ def get_atom_string(atoms, locs):
       str += "%s %4f %4f %4f; "%((atom,) + tuple(loc) )
     return atom_string, str
 
-
-NUM_TILES = jax.devices("ipu")[0].num_tiles
+try:
+    NUM_TILES = jax.devices("ipu")[0].num_tiles
+except:
+    NUM_TILES = 1472

```

Comment:
I would prefer to move this declaration inside a function only called on IPU.  Try/except at top level is a bit allergenic for me....

Reply:
Same...and I can't recall what led me to do it this way.